src/frontend/web_application/src/modules/device/services/signature/index.js

```diff
@@ -0,0 +1,27 @@
+import base64 from 'base64-js';
+import hash from 'object-hash';
+import { getKeypair, sign } from '../ecdsa';
+import { getConfig } from '../storage';
+import { buildURL } from '../../../../services/url';
+
+const buildMessage = ({ url, params }) => {
+  const message = buildURL(url, params);
+
+  return hash.sha1(message);
+};
+
+export const signRequest = (messageParts, priv) => {
```

Comment:
about naming thing, I like when it is one-line self-explanatory:

I'm not sure about `messageParts`, it looks like it the request object from axios but not sure. Actually I don't get why `req` becomes `messageParts`.

And  `priv` is a but too generic imo